src/huggingface_hub/file_download.py

```diff
@@ -961,6 +961,31 @@ def repo_folder_name(*, repo_id: str, repo_type: str) -> str:
     return REPO_ID_SEPARATOR.join(parts)
 
 
+def _check_disk_space(expected_size: Optional[int], target_dir: Union[str, Path]) -> None:
+    """Check disk usage and log a warning if there is not enough disk space to download the file.
+
+    Args:
+        expected_size (`int`, *optional*):
+            The expected size of the file in bytes.
+        target_dir (`str`):
+            The directory where the file will be stored after downloading.
+    """
+
+    target_dir = str(target_dir)
+    target_dir_free = shutil.disk_usage(target_dir).free
+
+    has_enough_space = target_dir_free >= expected_size
+
+    if not has_enough_space:
+        logger.warning(
+            "Not enough free disk space to download the file. "
+            f"The expected file size is: {expected_size / 1e6:.2f} MB. "
+            f"The target location {target_dir} only has {target_dir_free / 1e6:.2f} MB free space."
+        )
+
+    return has_enough_space
```

Comment:
```suggestion
```

No need for a return value. I believe you added this return value to make your tests pass. A better way to do this is to use [`self.assertWarns`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns) when you expect a warning to be raised. There are other occurrences of this in the tests if you want to have a look.